enclave/core/sgx/calls.c

```diff
@@ -135,12 +136,58 @@ extern bool oe_disable_debug_malloc_check;
 **==============================================================================
 */
 
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+extern oe_eeid_t* oe_eeid;
+extern const oe_enclave_config_t* oe_enclave_config;
+
+static oe_result_t _load_config(const oe_enclave_config_t* config)
+{
+    if (config && config->data && config->size)
+    {
+        if (oe_is_within_enclave(config->data, config->size) ||
```

Comment:
This check needs to be done on the enclave copy of config. Otherwise the host could exploit the TOCTOU vulnerability.

Reply:
Both are checked. This line of code checks that the incoming config is outside of the enclave (it aborts if within), a few lines below there's a check on the copy.

Reply:
> Both are checked. This line of code checks that the incoming config is outside of the enclave (it aborts if within), a few lines below there's a check on the copy.

The issue is that after line `147` that checks `config->data` before the control reaches line `159`, the host can change the value of `config->data`. So the `memcpy(copy->data, config->data, config->size)` could be manipulated to copy data from `config->data` that has been changed to point to within the enclave.

Reply:
Note that `oe_is_within_enclave` is not the same thing as `!oe_is_outside_enclave`. If a part of `config->data` lies within the enclave while the rest lies outside, then `oe_is_within_enclave` will return false.

Reply:
Yes, I see what you mean now. I will remove this check as it's really not necessary at all. I don't think there's a security problem with the data already being in the enclave address space.